Makefile

```diff
@@ -63,6 +63,10 @@ test: ## Run tests and collect coverage data
 	uv run coverage run -m pytest
 	@uv run coverage report
 
+.PHONY: test-fast
+test-fast: ## Same as test except no coverage. ~1/4th the time depending on hardware.
```

Comment:
We can add coverage in a next step tho.

I guess the motivation is that the CI is too slow, but I think we should also probably figure it out what is being slow.

Reply:
Ah, I see the changes you did.

I think it is worth it, but we can start with this change. :+1:

Reply:
I did a bunch of testing including using pytest-cov instead of coverage with xdist. The speed increase after adding pytest-cov was dramatically lower and made the complexity not worth it IMO

Reply:
I was also considering adding a test-all-python-fast.

Reply:
I don't think we need `pytest-cov`.